extensions/rich_text_components/Skillreview/directives/oppia-noninteractive-skillreview.directive.ts

```diff
@@ -82,6 +88,14 @@ angular.module('oppia').directive('oppiaNoninteractiveSkillreview', [
               }
             });
           };
+          // Function to check whether the RTE modal or the concept card
+          // modal should be triggered.
+          ctrl.toggleModal = function(event) {
```

Comment:
Why is this needed to be tied to ctrl?

Reply:
Because the function is being run within the scope of the directive. ctrl here refers to 'this'.

Reply:
Generally, we attach function/variable to ctrl only if they are going to be used in the template. Can we do `var <functionName> ...` here?

Reply:
@ARKEOLOGIST PTAL!